### ISSUE TITLE
Support Nullmarked on module-info (JPMS)

### ISSUE THREAD

**Issue by xenoterracide** [uber/NullAway#1083]
If a module-info exists and it has the `Nullmarked` annotation, all packages contained within should be treated as though they are also `Nullmarked` unless they are `NullUnmarked`, and etc... you get it. I know this behavior is desired, so this is mostly just so it can be tracked.

**Comment by msridhar** [comment]
We should and will eventually support this.  But generally speaking, at this point it doesn't seem like a library can rely just on annotating `module-info` to get the desired behavior; see https://github.com/jspecify/jspecify/issues/496#issuecomment-2532696315

**Comment by xenoterracide** [comment]
JPMS, getting in its own way for a decade, and not being helped along by the community :/

**Comment by marcphilipp** [comment]
We'd also like to use this in the JUnit build (see https://github.com/junit-team/junit5/pull/4557#issuecomment-2893541447).

**Comment by msridhar** [comment]
@marcphilipp thanks for the feedback!  We can try to prioritize adding this support.  Does the issue outlined at https://github.com/jspecify/jspecify/issues/496#issuecomment-2532696315 impact your thinking about using module-level `@NullMarked` annotations?  Just curious.

**Comment by marcphilipp** [comment]
Thanks for the pointer! I wasn't aware of the issue and thought that NullAway would always see module-level annotations regardless whether the jar is on the classpath or module path. IIUC the status quo is that downstream users of NullAway that have JUnit on the classpath will not see JUnit's packages as being `@NullMarked` because the module descriptor is ignored, correct?

**Comment by msridhar** [comment]
Yes that is correct. FYI @cpovirk we may want to document more prominently?

**Comment by cpovirk** [comment]
We probably should, thanks :(

Maybe @kevinb9n has an opinion on whether it's worth mentioning in [the Javadoc for `@NullMarked`](https://jspecify.dev/docs/api/org/jspecify/annotations/NullMarked.html), which is already home to at least one warning, that one about annotations at the _package_ level?
